src/main/java/com/iexec/common/docker/client/DockerClientInstance.java

```diff
@@ -0,0 +1,663 @@
+            log.info("Pulled docker image [name:{}]", imageName);
+            return true;
+        } catch (Exception e) {
+            log.error("Error pulling docker image [name:{}]", imageName, e);
+            return false;
+        }
+    }
+
+    public String getImageId(String imageName) {
+        if (StringUtils.isBlank(imageName)) {
+            return "";
+        }
+        String sanitizedImageName = sanitizeImageName(imageName);
+
+        try (ListImagesCmd listImagesCmd = this.client.listImagesCmd()) {
+            return listImagesCmd
+                    .withDanglingFilter(false)
+                    .withImageNameFilter(sanitizedImageName)
+                    .exec()
+                    .stream()
+                    .filter(image -> image.getRepoTags() != null && image.getRepoTags().length > 0)
+                    .filter(image -> Arrays.asList(image.getRepoTags()).contains(sanitizedImageName))
+                    .map(Image::getId)
+                    .findFirst()
+                    .orElse("");
+        } catch (Exception e) {
+            log.error("Error getting docker image id [name:{}]", imageName, e);
+        }
+        return "";
+    }
+
+    public String sanitizeImageName(String image) {
+        List<String> regexList = Arrays.asList( // order matters
+                "docker.io/library/(.*)", // docker.io/library/alpine:latest
+                "library/(.*)", // library/alpine:latest
+                "docker.io/(.*)", // docker.io/repo/image:latest
+                "docker.com/(.*)"); // registry.hub.docker.com/repo/image:latest (deprecated)
+
+        for (String regex : regexList) {
+            Matcher m = Pattern.compile(regex).matcher(image);
+            if (m.find()) {
+                return m.group(1);
+            }
+        }
+        return image;
+    }
+
+    public boolean removeImage(String imageName) {
+        return removeImageById(getImageId(imageName));
+    }
+
+    // TODO remove image by name to avoid conflict error
+    /*
+    * If same image has been pulled with multiple repository base names:
+    *
+    * - docker.io/repo/image:latest
+    * and
+    * - registry.hub.docker.com/repo/image:latest (e.g)
+    *
+    * It will return conflict:
+    * unable to delete <imageId> (must be forced) - image is referenced in multiple repositories
+    * */
+    boolean removeImageById(String imageId) {
+        if (StringUtils.isBlank(imageId)) {
+            return false;
+        }
+        try (RemoveImageCmd removeImageCmd =
+                     this.client.removeImageCmd(imageId)) {
+            removeImageCmd.exec();
+            return true;
+        } catch (Exception e) {
+            log.error("Error removing docker image by id [id:{}]", imageId, e);
+            return false;
+        }
+    }
+
+    /**
+     * Docker container
+     * 
+     */
+
+    public DockerRunResponse run(DockerRunRequest dockerRunRequest) {
```

Comment:
Will be hard to test

Reply:
I agree but we don't have a choice for now.
PS: tests coming in the next PR.